Brokerages/Exante/ExanteBrokerage.DataQueueHandler.cs

```diff
@@ -0,0 +1,249 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Interfaces;
+using System.Linq;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Exante.Net.Enums;
+using Exante.Net.Objects;
+using QuantConnect.Data;
+using QuantConnect.Packets;
+using NodaTime;
+using QuantConnect.Data.Market;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+
+namespace QuantConnect.Brokerages.Exante
+{
+    public partial class ExanteBrokerage : IDataQueueHandler
+    {
+        private readonly ConcurrentDictionary<string, Symbol> _subscribedTickers =
+            new ConcurrentDictionary<string, Symbol>();
+
+        private readonly ConcurrentDictionary<string, (ExanteStreamSubscription, ExanteStreamSubscription)>
+            _subscribedTickersStreamSubscriptions =
+                new ConcurrentDictionary<string, (ExanteStreamSubscription, ExanteStreamSubscription)>();
+
+        /// <inheritdoc />
+        public IEnumerator<BaseData> Subscribe(SubscriptionDataConfig dataConfig, EventHandler newDataAvailableHandler)
+        {
+            if (!CanSubscribe(dataConfig.Symbol))
+            {
+                return Enumerable.Empty<BaseData>().GetEnumerator();
+            }
+
+            var enumerator = Aggregator.Add(dataConfig, newDataAvailableHandler);
+            _subscriptionManager.Subscribe(dataConfig);
+
+            return enumerator;
+        }
+
+        /// <summary>
+        /// Returns true if this data provide can handle the specified symbol
+        /// </summary>
+        /// <param name="symbol">The symbol to be handled</param>
+        /// <returns>True if this data provider can get data for the symbol, false otherwise</returns>
+        private bool CanSubscribe(Symbol symbol)
+        {
+            var supportedSecurityTypes = new HashSet<SecurityType>
+            {
+                SecurityType.Forex,
+                SecurityType.Equity,
+                SecurityType.Future,
+                SecurityType.Option,
+                SecurityType.Cfd,
+                SecurityType.Index,
+                SecurityType.Crypto,
+            };
+
+            // ignore unsupported security types
+            if (!supportedSecurityTypes.Contains(symbol.ID.SecurityType))
+            {
+                return false;
+            }
+
+            // ignore universe symbols
+            return !symbol.Value.Contains("-UNIVERSE-");
+        }
+
+        /// <inheritdoc />
+        public void Unsubscribe(SubscriptionDataConfig dataConfig)
+        {
+            _subscriptionManager.Unsubscribe(dataConfig);
+            Aggregator.Remove(dataConfig);
+        }
+
+        /// <inheritdoc />
```

Comment:
Our code are docs by policy we fully implement all comments; please copy paste if required but fully implement each XML block.